src/aarch64.cpp

```diff
@@ -0,0 +1,340 @@
+    }));
+    instrucion_set[A64_B_GT] = arg_indexed_array<binatr>(binatr({ //TODO(ch): there is no B_GT instruction in ARM processors, it's prespecialized version of B.cond. We must make switchers much more flexible and functional to support real B.cond. Specialization is: fixed condition.
+        binatr::detail(binatr::detail::D_STATIC, 0x54,8),//01010100
+        binatr::detail(binatr::detail::D_OFFSET, 19)    ,//jump offset
+        binatr::detail(binatr::detail::D_STATIC, 0xC,5)  //01100
+    }));
+    instrucion_set[A64_B_GE] = arg_indexed_array<binatr>(binatr({ //TODO(ch): there is no B_GE instruction in ARM processors, it's prespecialized version of B.cond. We must make switchers much more flexible and functional to support real B.cond. Specialization is: fixed condition.
+        binatr::detail(binatr::detail::D_STATIC, 0x54,8),//01010100
+        binatr::detail(binatr::detail::D_OFFSET, 19)    ,//jump offset
+        binatr::detail(binatr::detail::D_STATIC, 0xA,5)  //01010
+    }));
+    instrucion_set[A64_RET] = arg_indexed_array<binatr>(binatr({ //TODO(ch): It's good for macos, but i'm not sure, it's not defined by calling convention. Must check.
+        binatr::detail(binatr::detail::D_STATIC, 0x3597C0,22),//1101011001011111000000
+        binatr::detail(binatr::detail::D_REG, 5)             ,//reg register
+        binatr::detail(binatr::detail::D_STATIC, 0x0,5)       //00000
+    }));
+};
+
+inline m2b_map& get_instrucion_set()
+{
+    if(instrucion_set.empty())
+        init_instrucion_set();
+    return instrucion_set;
+}
+
+m2m_map target_mnemonics;
+void init_target_mnemonics()
+{
+    {//OP_LOAD
+        mnemotr loadbase(A64_LDRSW, {
+            mnemotr::argutr(Arg(0)),
+            mnemotr::argutr(size_t(2)),
+            mnemotr::argutr(size_t(0))
+            });
+        mnemotr loadoffset(A64_LDRSW, {
+            mnemotr::argutr(size_t(3)),
+            mnemotr::argutr(size_t(2)),
+            mnemotr::argutr(size_t(0))
+            });
+        arg_indexed_array<mnemotr> load_vars;
+        load_vars.m_condition = arg_indexed_array<binatr>::C_ARRSIZE;
+        load_vars.m_branches[3] = arg_indexed_array<mnemotr>(loadbase);
+        load_vars.m_branches[4] = arg_indexed_array<mnemotr>(loadoffset);
+        target_mnemonics[OP_LOAD] = load_vars;
+    }
+
+    //TODO(ch): Add branch check: op.args[0].value must be equal to TYPE_I32. Same for many and many and many operations.
+    target_mnemonics[OP_STORE] = arg_indexed_array<mnemotr>( mnemotr(A64_STR32_I, {
+        mnemotr::argutr(Arg(0)),
+        mnemotr::argutr(size_t(1)),
+        mnemotr::argutr(size_t(2))
+    }));
+
+    target_mnemonics[OP_MOV] = arg_indexed_array<mnemotr>( mnemotr(A64_MOV, {
+        mnemotr::argutr(size_t(1)),
+        mnemotr::argutr(size_t(0))
+    }) );
+
+    target_mnemonics[OP_AUG_ADD] = arg_indexed_array<mnemotr>( mnemotr(A64_ADD_R, {
+        mnemotr::argutr(size_t(1)),
```

Comment:
can we handle augmentation operations earlier in the compiler pipeline? why not just translate `y += x` to `y = y + x` bytecode and forget completely about OP_AUG_ADD?

Reply:
Done.